lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -867,10 +867,22 @@ getConformanceAndConcreteType(ASTContext &Ctx,
   // opened existential type, so we must keep track of the original
   // defining instruction.
   if (ConcreteType->isOpenedExistential()) {
-    assert(!InitExistential->getTypeDependentOperands().empty() &&
-           "init_existential is supposed to have a typedef operand");
-    ConcreteTypeDef = cast<SingleValueInstruction>(
-      InitExistential->getTypeDependentOperands()[0].get());
+    if (InitExistential->getTypeDependentOperands().empty()) {
+      auto op = InitExistential->getOperand(0);
+      assert(op->getType().hasOpenedExistential() &&
+             "init_existential is supposed to have a typedef operand");
+      ConcreteTypeDef = cast<SingleValueInstruction>(op);
+    } else {
+      ConcreteTypeDef = cast<SingleValueInstruction>(
+          InitExistential->getTypeDependentOperands()[0].get());
+    }
+  }
+
+  // If we can't find the ExactConformance - bail
+  // We only encountered such situations in hand-written SIL (so far)
+  // Should take care of it nonetheless because to avoid a compiler crash
+  if (!ExactConformance.hasValue()) {
```

Comment:
I think it's fine to assert on invalid SIL, even when hand-written. I'd rather shake out the cases where SILGen generates invalid SIL than have this

Reply:
Fixed